File: npm/ng-packs/packages/cms/src/lib/components/admin/fields/create-or-update-field-input-base.ts
import { FormGroup, Validators } from "@angular/forms";

export class CreateOrUpdateFieldInputBase {
    /**分组id */
    groupId: any = ['', []];

    /**字段名称 Display name of this field */
    displayName: any = ['', [Validators.required]];

    /**字段唯一名称 Unique Name*/
    name: any = ['', [Validators.required]];

    /**描述 说明 */
    description: any = ['', []];

    /**FieldType字段类型 表单控件名称 */
    formControlName: any = [undefined, [Validators.required]];

    /**动态表单配置 */
    formConfiguration: FormGroup | undefined = new FormGroup({})


    constructor(data?: CreateOrUpdateFieldInputBase) {
        if (data) {
            for (const key in data) {
                if (data.hasOwnProperty(key)) {
                    this[key] = data[key];
                }
            }
        }

    }
}

File: modules/cms/src/Dignite.Cms.Admin.Application.Contracts/Dignite/Cms/Admin/Fields/CreateOrUpdateFieldInputBase.cs
using Dignite.Abp.DynamicForms;
using Dignite.Cms.Fields;
using JetBrains.Annotations;
using System;
using System.ComponentModel.DataAnnotations;
using Volo.Abp.Validation;

namespace Dignite.Cms.Admin.Fields
{
    public abstract class CreateOrUpdateFieldInputBase
    {
        protected CreateOrUpdateFieldInputBase()
        {
            FormConfiguration=new FormConfigurationDictionary();
        }

        public  Guid? GroupId { get; set; }

        /// <summary>
        /// Display name of this field.
        /// </summary>
        [Required]
        [DynamicMaxLength(typeof(FieldConsts), nameof(FieldConsts.MaxDisplayNameLength))]
        public string DisplayName { get; set; }

        /// <summary>
        /// Unique Name
        /// </summary>
        [Required]
        [DynamicMaxLength(typeof(FieldConsts), nameof(FieldConsts.MaxNameLength))]
        [RegularExpression(FieldConsts.NameRegularExpression)]
        public string Name { get; set; }

        /// <summary>
        /// Description of the field.
        /// </summary>
        [CanBeNull]
        [DynamicMaxLength(typeof(FieldConsts), nameof(FieldConsts.MaxDescriptionLength))]
        public string Description { get; set; }

        [Required]
        [DynamicMaxLength(typeof(FieldConsts), nameof(FieldConsts.MaxFormControlNameLength))]
        public string FormControlName { get; set; }

        [Required]
        public FormConfigurationDictionary FormConfiguration { get; set; }

    }
}
